### PR TITLE
make navbar vue element

### PR DESCRIPTION
**Pull request by heatherbooker**
@dharness

### CODE REVIEW

File: public/javascripts/nav.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$(document).ready(function() {
```
Comment (dharness) [inline public/javascripts/nav.js]: We are going to want to get rid of these suckers

File: routes/index.js [hunk 2]
```diff
@@ -6,13 +6,16 @@ router.get('/', function(req, res, next) {
   res.render('index', {
     title: 'chubbo-chat',
     //pass src for page bkground only needed on landing
-    background: '/javascripts/floatingDots.js'
+    script: '/javascripts/floatingDots.js'
```
Comment (dharness) [inline routes/index.js]: Why did you add this as a variable instead of hardcoding it in the .pug file? What value does this method add?

File: routes/index.js [hunk 3]
```diff
@@ -6,13 +6,16 @@ router.get('/', function(req, res, next) {
...
   });
 });
 
 // GET dashboard (for logged in users)
 router.get('/dashboard', function(req, res, next) {
-  res.render('dashboard', {title: 'chubbo-chat: dashboard'});
+  res.render('dashboard', {
+    title: 'chubbo-chat: dashboard',
+    script: '/javascripts/dashboard.js'
```
Comment (dharness) [inline routes/index.js]: Again

File: views/layout.pug [hunk 4]
```diff
@@ -21,7 +21,17 @@ html
       firebase.initializeApp(config);
     script(src='https://ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js')
     script(src='https://npmcdn.com/vue/dist/vue.js')
-    script(src=background)
+    script(src=script)
     script(src='/javascripts/login.js')
-    script(src='/javascripts/dashboard.js')
+    script(src='/javascripts/nav.js')
+    block nav
```
Comment (dharness) [inline views/layout.pug]: So yea, this nav code should be it's own component

File: routes/index.js (hunk 2 above)
Comment (heatherbooker) [inline routes/index.js]: I did it like this because on the dashboard page it gets an error, since the element its looking for only exists on the landing page: http://puu.sh/pUPyI/b17cf9a152.png 
But I guess it doesn't really matter if that error comes up, it doesn't affect anything

**Comment by heatherbooker** [comment]
@dharness what do you think of updates?

File: public/javascripts/nav.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+var navbar = Vue.extend({
+  props: ['logo', 'loginBtn', 'logoutBtn'],
+  template: `
+    <div class="container-fluid">
+      <div class="row cc-navbar">
+        <div class="col-xs-8"><a href="#">
+            <h1 class="cc-logo">{{ logo }}</h1></a></div>
```
Comment (dharness) [inline public/javascripts/nav.js]: You don't need to make this a prop, it's never going to change

File: public/javascripts/nav.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        <div class="col-xs-4">
```
Comment (dharness) [inline public/javascripts/nav.js]: Let's do this with a v-if statement instead

File: public/javascripts/vueInstance.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+var vueInstance = new Vue({
```
Comment (dharness) [inline public/javascripts/vueInstance.js]: this isn't just _any_ vue instance - it is the nav. You should call it that

**Comment by heatherbooker** [comment]
@dharness cleaned er up, CIGAR?

File: public/javascripts/login.js [hunk 8]
```diff
@@ -1,54 +1,44 @@
-$(document).ready(function() {
-
+window.onload = function() {
```
Comment (dharness) [inline public/javascripts/login.js]: I don't think you need a window.onload for this

File: public/javascripts/login.js [hunk 9]
```diff
@@ -1,54 +1,44 @@
...
+    this.toggleLoginBtn(true);
   } else {
-    toggleLoginBtn(false);
+    this.toggleLoginBtn(false);
   }
 }
 
-
-function toggleLoginBtn(isLoggedIn) {
+Login.prototype.toggleLoginBtn = function(isLoggedIn) {
```
Comment (dharness) [inline public/javascripts/login.js]: What is this used for? Why can't we do this with vue.js?

File: public/javascripts/main.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+var Main = Vue.extend({
+  template: `
+    <div>
+      <nav-bar></nav-bar>
+      <router-view></router-view>
+    </div>
+  `,
+  components: {
+    'nav-bar': navbar
```
Comment (dharness) [inline public/javascripts/main.js:13]: nicely done.

File: public/javascripts/login.js (hunk 8 above)
Comment (heatherbooker) [inline public/javascripts/login.js]: that's the way firebase suggests, and i tried removing it but the other methods fail / throw errors

File: public/javascripts/login.js (hunk 9 above)
Comment (heatherbooker) [inline public/javascripts/login.js]: it's called when the page first loads, to check whether or not a user is already logged in

Comment (dharness) [inline public/javascripts/login.js]: Why can't we do this with vue.js?

File: public/javascripts/login.js (hunk 8 above)
Comment (dharness) [inline public/javascripts/login.js]: I think you can use vue.js hooks instead

Comment (heatherbooker) [inline public/javascripts/login.js]: awesome thanks for the tip

**Comment by dharness** [comment]
LGTM
